torch/csrc/distributed/rpc/unpickled_python_call.cpp

```diff
@@ -14,13 +14,22 @@ UnpickledPythonCall::UnpickledPythonCall(
   pythonUdf_ = pythonRpcHandler.deserialize(serializedPyObj);
 }
 
+UnpickledPythonCall::~UnpickledPythonCall() {
+  // explicitly setting PyObject* to nullptr to prevent py::object's dtor to
+  // decref on the PyObject again.
+  // See Note [Destructing py::object] in python_ivalue.h
+  py::gil_scoped_acquire acquire;
+  pythonUdf_.dec_ref();
+  pythonUdf_.ptr() = nullptr;
+}
```

Comment:
By doing this, we can also remove the GIL on line 176 (under MessageType::PYTHON_CALL) and line 287 (under MessageType::PYTHON_REMOTE_CALL).

Reply:
We still need the GIL there, because `runPythonUdf(...)` returns a new `py::object` and we need the GIL to guard its destruction. Let me add a comment there.